RecoLocalMuon/GEMSegment/plugins/GEMSegmentAlgorithm.h

```diff
@@ -0,0 +1,39 @@
+#ifndef GEMRecHit_GEMSegmentAlgorithm_h
+#define GEMRecHit_GEMSegmentAlgorithm_h
+
+/** \class GEMSegmentAlgo derived from CSC
+ * An abstract base class for algorithmic classes used to
+ * build segments in one ensemble of GEM detector 
+ *
+ * Implementation notes: <BR>
+ * For example, GEMSegmAlgoPV inherits from this class,
+ *
+ * \author Piet Verwilligen
+ *
+ */
+
+#include "DataFormats/GEMRecHit/interface/GEMRecHitCollection.h"
+#include "DataFormats/GEMRecHit/interface/GEMSegment.h"
+#include "Geometry/GEMGeometry/interface/GEMSuperChamber.h"
+#include "Geometry/GEMGeometry/interface/GEMEtaPartition.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include <map>
+#include <vector>
+
+class GEMSegmentAlgorithm {
```

Comment:
... and in this case the GEMSegAlgo should become GEMSegmentAlgorithm for the lack of a better specification, since it's the first derived version of this abstract type